scout/adapter/mongo/case.py

```diff
@@ -508,8 +502,11 @@ def _conditional_set_query_value(query, condition, set_key, set_value):
         if yield_query:
             return query
 
-        if order:
-            return self.case_collection.find(query, projection)
+        if name_query and query_field in ["similar_case", "similar_pheno"]:
+            result_order: list = query["_id"]["$in"]
+            results = self.case_collection.find(query, projection)
+            # Return the result in order of ascending phenotype similarity (the same order or the _ids provided in the query)
+            return sorted(list(results), key=lambda res: result_order.index(res["_id"]))
```

Comment:
😆

Reply:
This is the fix: since the order of the _ids in the query (`{"_id": {"$in": [list of _ids]}`) has the ones that are most similar first, so you just need to sort by this list

Reply:
Nice, and I do like this kind of lambda use! 😁